charmcraft/commands/store/store.py

```diff
@@ -315,6 +315,9 @@ def _upload(self, endpoint, filepath, *, extra_fields=None):
     def upload(self, name, filepath):
         """Upload the content of filepath to the indicated charm."""
         endpoint = f"/v1/charm/{name}/revisions"
+        if not self.whoami():
+            raise CraftError("Existing credentials are no longer valid for Charmhub.")
+
```

Comment:
Based on the tests, couldn't we just do:

```suggestion
        # This raises an exception if the user's credentials are invalid.
        self.whoami()
```

I can't see a case where `whoami` would return some falsey value rather than raising an exception.

Reply:
I can't see it either, but I put it here just in case.